From 6df8e0a26e01169ac4bf6298216b0afa936edcc9 Mon Sep 17 00:00:00 2001
From: Martin Kuhn <martin@kuhn.host>
Date: Sat, 19 Sep 2026 17:18:57 +0200
Subject: [PATCH 1/3] Patched for Raspberry Pi 5

Adjust the arm64 kernel module list to the RPi5 kernel config (modules built in by the pkgs config patch removed, extra modules added).
---
 hack/modules-arm64.txt | 190 ++++++++++++++++++-----------------------
 1 file changed, 81 insertions(+), 109 deletions(-)

diff --git a/hack/modules-arm64.txt b/hack/modules-arm64.txt
index 35656feef..089768db8 100644
--- a/hack/modules-arm64.txt
+++ b/hack/modules-arm64.txt
@@ -4,20 +4,18 @@ kernel/crypto/async_tx/async_pq.ko
 kernel/crypto/async_tx/async_raid6_recov.ko
 kernel/crypto/async_tx/async_tx.ko
 kernel/crypto/async_tx/async_xor.ko
+kernel/crypto/blake2b_generic.ko
 kernel/crypto/hkdf.ko
 kernel/crypto/xor.ko
+kernel/crypto/xxhash_generic.ko
+kernel/crypto/zstd.ko
 kernel/drivers/acpi/ac.ko
 kernel/drivers/acpi/battery.ko
 kernel/drivers/acpi/button.ko
 kernel/drivers/acpi/fan.ko
 kernel/drivers/acpi/thermal.ko
 kernel/drivers/acpi/video.ko
-kernel/drivers/ata/ahci.ko
 kernel/drivers/ata/ata_piix.ko
-kernel/drivers/ata/pata_amd.ko
-kernel/drivers/ata/pata_marvell.ko
-kernel/drivers/ata/pata_oldpiix.ko
-kernel/drivers/ata/pata_sch.ko
 kernel/drivers/block/nbd.ko
 kernel/drivers/block/ublk_drv.ko
 kernel/drivers/block/virtio_blk.ko
@@ -26,57 +24,38 @@ kernel/drivers/char/ipmi/ipmi_si.ko
 kernel/drivers/char/virtio_console.ko
 kernel/drivers/crypto/tegra/tegra-se.ko
 kernel/drivers/crypto/virtio/virtio_crypto.ko
+kernel/drivers/dma/bcm-sba-raid.ko
+kernel/drivers/gpu/drm/amd/amdgpu/amdgpu.ko
+kernel/drivers/gpu/drm/amd/amdxcp/amdxcp.ko
 kernel/drivers/gpu/drm/drm_buddy.ko
 kernel/drivers/gpu/drm/drm_exec.ko
-kernel/drivers/gpu/drm/drm_gpuvm.ko
 kernel/drivers/gpu/drm/drm_panel_backlight_quirks.ko
 kernel/drivers/gpu/drm/drm_suballoc_helper.ko
 kernel/drivers/gpu/drm/drm_ttm_helper.ko
 kernel/drivers/gpu/drm/drm_vram_helper.ko
 kernel/drivers/gpu/drm/hisilicon/hibmc/hibmc-drm.ko
+kernel/drivers/gpu/drm/panfrost/panfrost.ko
 kernel/drivers/gpu/drm/scheduler/gpu-sched.ko
 kernel/drivers/gpu/drm/tegra/tegra-drm.ko
 kernel/drivers/gpu/drm/ttm/ttm.ko
+kernel/drivers/gpu/drm/v3d/v3d.ko
+kernel/drivers/gpu/drm/vc4/vc4.ko
 kernel/drivers/gpu/drm/virtio/virtio-gpu.ko
 kernel/drivers/gpu/host1x/host1x.ko
-kernel/drivers/hid/hid-a4tech.ko
-kernel/drivers/hid/hid-apple.ko
-kernel/drivers/hid/hid-belkin.ko
-kernel/drivers/hid/hid-cherry.ko
-kernel/drivers/hid/hid-chicony.ko
-kernel/drivers/hid/hid-cypress.ko
-kernel/drivers/hid/hid-ezkey.ko
-kernel/drivers/hid/hid-gyration.ko
-kernel/drivers/hid/hid-ite.ko
-kernel/drivers/hid/hid-kensington.ko
 kernel/drivers/hid/hid-lg-g15.ko
 kernel/drivers/hid/hid-logitech.ko
-kernel/drivers/hid/hid-microsoft.ko
-kernel/drivers/hid/hid-monterey.ko
 kernel/drivers/hid/hid-multitouch.ko
-kernel/drivers/hid/hid-petalynx.ko
-kernel/drivers/hid/hid-pl.ko
 kernel/drivers/hid/hid-redragon.ko
-kernel/drivers/hid/hid-samsung.ko
-kernel/drivers/hid/hid-sunplus.ko
-kernel/drivers/hid/hid-topseed.ko
-kernel/drivers/hid/usbhid/usbhid.ko
 kernel/drivers/hv/hv_balloon.ko
 kernel/drivers/hv/hv_utils.ko
 kernel/drivers/hwmon/acpi_power_meter.ko
 kernel/drivers/hwmon/drivetemp.ko
 kernel/drivers/hwmon/i5k_amb.ko
 kernel/drivers/hwmon/spd5118.ko
-kernel/drivers/i2c/algos/i2c-algo-bit.ko
-kernel/drivers/i2c/busses/i2c-i801.ko
-kernel/drivers/i2c/i2c-mux.ko
 kernel/drivers/infiniband/core/ib_cm.ko
 kernel/drivers/infiniband/core/ib_core.ko
-kernel/drivers/infiniband/core/ib_umad.ko
-kernel/drivers/infiniband/core/ib_uverbs.ko
 kernel/drivers/infiniband/core/iw_cm.ko
 kernel/drivers/infiniband/core/rdma_cm.ko
-kernel/drivers/infiniband/core/rdma_ucm.ko
 kernel/drivers/infiniband/hw/bnxt_re/bnxt_re.ko
 kernel/drivers/infiniband/hw/hns/hns-roce-hw-v2.ko
 kernel/drivers/infiniband/hw/irdma/irdma.ko
@@ -84,6 +63,7 @@ kernel/drivers/infiniband/hw/mlx4/mlx4_ib.ko
 kernel/drivers/infiniband/hw/mlx5/mlx5_ib.ko
 kernel/drivers/infiniband/sw/rxe/rdma_rxe.ko
 kernel/drivers/infiniband/ulp/ipoib/ib_ipoib.ko
+kernel/drivers/input/misc/uinput.ko
 kernel/drivers/input/misc/xen-kbdfront.ko
 kernel/drivers/input/mouse/psmouse.ko
 kernel/drivers/input/serio/hyperv-keyboard.ko
@@ -91,7 +71,6 @@ kernel/drivers/iommu/iommufd/iommufd.ko
 kernel/drivers/irqchip/irq-imx-mu-msi.ko
 kernel/drivers/leds/led-class-multicolor.ko
 kernel/drivers/mailbox/bcm-flexrm-mailbox.ko
-kernel/drivers/md/bcache/bcache.ko
 kernel/drivers/md/dm-bio-prison.ko
 kernel/drivers/md/dm-cache-smq.ko
 kernel/drivers/md/dm-cache.ko
@@ -103,32 +82,23 @@ kernel/drivers/md/dm-service-time.ko
 kernel/drivers/md/dm-thin-pool.ko
 kernel/drivers/md/persistent-data/dm-persistent-data.ko
 kernel/drivers/md/raid456.ko
-kernel/drivers/misc/hpilo.ko
-kernel/drivers/mmc/host/sdhci-acpi.ko
-kernel/drivers/mmc/host/sdhci-brcmstb.ko
-kernel/drivers/mmc/host/sdhci-cadence.ko
-kernel/drivers/mmc/host/sdhci-iproc.ko
-kernel/drivers/mmc/host/sdhci-msm.ko
-kernel/drivers/mmc/host/sdhci-of-arasan.ko
-kernel/drivers/mmc/host/sdhci-of-dwcmshc.ko
-kernel/drivers/mmc/host/sdhci-of-esdhc.ko
-kernel/drivers/mmc/host/sdhci-pci.ko
-kernel/drivers/mmc/host/sdhci-pltfm.ko
-kernel/drivers/mmc/host/sdhci-tegra.ko
-kernel/drivers/mmc/host/sdhci-uhs2.ko
-kernel/drivers/mmc/host/sdhci-xenon-driver.ko
-kernel/drivers/mmc/host/sdhci.ko
-kernel/drivers/mmc/host/sdhci_f_sdh30.ko
+kernel/drivers/media/common/uvc.ko
+kernel/drivers/media/common/videobuf2/videobuf2-common.ko
+kernel/drivers/media/common/videobuf2/videobuf2-memops.ko
+kernel/drivers/media/common/videobuf2/videobuf2-v4l2.ko
+kernel/drivers/media/common/videobuf2/videobuf2-vmalloc.ko
+kernel/drivers/media/mc/mc.ko
+kernel/drivers/media/usb/uvc/uvcvideo.ko
+kernel/drivers/media/v4l2-core/v4l2-dv-timings.ko
+kernel/drivers/media/v4l2-core/videodev.ko
 kernel/drivers/net/ethernet/amazon/ena/ena.ko
 kernel/drivers/net/ethernet/amd/xgbe/amd-xgbe.ko
-kernel/drivers/net/ethernet/aquantia/atlantic/atlantic.ko
 kernel/drivers/net/ethernet/atheros/alx/alx.ko
-kernel/drivers/net/ethernet/broadcom/bnx2.ko
-kernel/drivers/net/ethernet/broadcom/bnx2x/bnx2x.ko
-kernel/drivers/net/ethernet/broadcom/bnxt/bnxt_en.ko
-kernel/drivers/net/ethernet/broadcom/tg3.ko
-kernel/drivers/net/ethernet/cavium/common/cavium_ptp.ko
-kernel/drivers/net/ethernet/cisco/enic/enic.ko
+kernel/drivers/net/ethernet/chelsio/cxgb/cxgb.ko
+kernel/drivers/net/ethernet/chelsio/cxgb3/cxgb3.ko
+kernel/drivers/net/ethernet/chelsio/cxgb4/cxgb4.ko
+kernel/drivers/net/ethernet/chelsio/cxgb4vf/cxgb4vf.ko
+kernel/drivers/net/ethernet/chelsio/inline_crypto/ch_ipsec/ch_ipsec.ko
 kernel/drivers/net/ethernet/google/gve/gve.ko
 kernel/drivers/net/ethernet/hisilicon/hip04_eth.ko
 kernel/drivers/net/ethernet/hisilicon/hisi_femac.ko
@@ -142,24 +112,8 @@ kernel/drivers/net/ethernet/hisilicon/hns3/hclgevf.ko
 kernel/drivers/net/ethernet/hisilicon/hns3/hnae3.ko
 kernel/drivers/net/ethernet/hisilicon/hns3/hns3.ko
 kernel/drivers/net/ethernet/hisilicon/hns_mdio.ko
-kernel/drivers/net/ethernet/intel/e100.ko
-kernel/drivers/net/ethernet/intel/e1000/e1000.ko
-kernel/drivers/net/ethernet/intel/e1000e/e1000e.ko
-kernel/drivers/net/ethernet/intel/i40e/i40e.ko
-kernel/drivers/net/ethernet/intel/iavf/iavf.ko
-kernel/drivers/net/ethernet/intel/ice/ice.ko
 kernel/drivers/net/ethernet/intel/idpf/idpf.ko
-kernel/drivers/net/ethernet/intel/igb/igb.ko
-kernel/drivers/net/ethernet/intel/igbvf/igbvf.ko
-kernel/drivers/net/ethernet/intel/igc/igc.ko
-kernel/drivers/net/ethernet/intel/ixgbe/ixgbe.ko
-kernel/drivers/net/ethernet/intel/ixgbevf/ixgbevf.ko
-kernel/drivers/net/ethernet/intel/libeth/libeth.ko
 kernel/drivers/net/ethernet/intel/libeth/libeth_xdp.ko
-kernel/drivers/net/ethernet/intel/libie/libie.ko
-kernel/drivers/net/ethernet/intel/libie/libie_adminq.ko
-kernel/drivers/net/ethernet/intel/libie/libie_fwlog.ko
-kernel/drivers/net/ethernet/marvell/sky2.ko
 kernel/drivers/net/ethernet/mellanox/mlx4/mlx4_core.ko
 kernel/drivers/net/ethernet/mellanox/mlx4/mlx4_en.ko
 kernel/drivers/net/ethernet/mellanox/mlx5/core/mlx5_core.ko
@@ -169,45 +123,52 @@ kernel/drivers/net/ethernet/mellanox/mlxsw/mlxsw_i2c.ko
 kernel/drivers/net/ethernet/mellanox/mlxsw/mlxsw_minimal.ko
 kernel/drivers/net/ethernet/mellanox/mlxsw/mlxsw_pci.ko
 kernel/drivers/net/ethernet/mellanox/mlxsw/mlxsw_spectrum.ko
-kernel/drivers/net/ethernet/qlogic/qed/qed.ko
-kernel/drivers/net/ethernet/qlogic/qede/qede.ko
-kernel/drivers/net/ethernet/qlogic/qlcnic/qlcnic.ko
-kernel/drivers/net/ethernet/realtek/8139too.ko
-kernel/drivers/net/ethernet/realtek/r8169.ko
 kernel/drivers/net/ethernet/sfc/sfc.ko
 kernel/drivers/net/ethernet/sfc/siena/sfc-siena.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-dwc-qos-eth.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-generic.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-imx.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-ipq806x.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-meson.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-meson8b.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.ko
 kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-renesas-gbeth.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-rk.ko
 kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-sun55i.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-sun8i.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-sunxi.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/dwmac-tegra.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/stmmac-pci.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/stmmac-platform.ko
-kernel/drivers/net/ethernet/stmicro/stmmac/stmmac.ko
 kernel/drivers/net/hyperv/hv_netvsc.ko
 kernel/drivers/net/ifb.ko
-kernel/drivers/net/mdio.ko
 kernel/drivers/net/mdio/mdio-i2c.ko
+kernel/drivers/net/mdio/mdio-mux-meson-gxl.ko
 kernel/drivers/net/pcs/pcs-rzn1-miic.ko
-kernel/drivers/net/pcs/pcs_xpcs.ko
-kernel/drivers/net/phy/dp83867.ko
+kernel/drivers/net/phy/ax88796b.ko
 kernel/drivers/net/phy/sfp.ko
-kernel/drivers/net/usb/r8152.ko
+kernel/drivers/net/thunderbolt/thunderbolt_net.ko
+kernel/drivers/net/usb/aqc111.ko
+kernel/drivers/net/usb/asix.ko
+kernel/drivers/net/usb/ax88179_178a.ko
+kernel/drivers/net/usb/cdc_eem.ko
+kernel/drivers/net/usb/cdc_ether.ko
+kernel/drivers/net/usb/cdc_mbim.ko
+kernel/drivers/net/usb/cdc_ncm.ko
+kernel/drivers/net/usb/cdc_subset.ko
+kernel/drivers/net/usb/ch9200.ko
+kernel/drivers/net/usb/cx82310_eth.ko
+kernel/drivers/net/usb/dm9601.ko
+kernel/drivers/net/usb/gl620a.ko
+kernel/drivers/net/usb/huawei_cdc_ncm.ko
+kernel/drivers/net/usb/int51x1.ko
+kernel/drivers/net/usb/kalmia.ko
+kernel/drivers/net/usb/lg-vl600.ko
+kernel/drivers/net/usb/mcs7830.ko
+kernel/drivers/net/usb/net1080.ko
+kernel/drivers/net/usb/plusb.ko
+kernel/drivers/net/usb/qmi_wwan.ko
+kernel/drivers/net/usb/r8153_ecm.ko
+kernel/drivers/net/usb/rndis_host.ko
+kernel/drivers/net/usb/sierra_net.ko
+kernel/drivers/net/usb/smsc75xx.ko
+kernel/drivers/net/usb/smsc95xx.ko
+kernel/drivers/net/usb/sr9700.ko
+kernel/drivers/net/usb/sr9800.ko
+kernel/drivers/net/usb/usbnet.ko
+kernel/drivers/net/usb/zaurus.ko
 kernel/drivers/net/virtio_net.ko
-kernel/drivers/net/vmxnet3/vmxnet3.ko
 kernel/drivers/net/vrf.ko
 kernel/drivers/net/xen-netfront.ko
 kernel/drivers/nvme/common/nvme-auth.ko
 kernel/drivers/nvme/host/nvme-rdma.ko
-kernel/drivers/nvme/host/nvme.ko
 kernel/drivers/nvme/target/nvme-loop.ko
 kernel/drivers/nvme/target/nvmet-fc.ko
 kernel/drivers/nvme/target/nvmet-rdma.ko
@@ -229,19 +190,16 @@ kernel/drivers/scsi/hisi_sas/hisi_sas_main.ko
 kernel/drivers/scsi/hisi_sas/hisi_sas_v1_hw.ko
 kernel/drivers/scsi/hisi_sas/hisi_sas_v2_hw.ko
 kernel/drivers/scsi/hisi_sas/hisi_sas_v3_hw.ko
-kernel/drivers/scsi/hpsa.ko
 kernel/drivers/scsi/hv_storvsc.ko
 kernel/drivers/scsi/libfc/libfc.ko
 kernel/drivers/scsi/lpfc/lpfc.ko
-kernel/drivers/scsi/megaraid/megaraid_sas.ko
 kernel/drivers/scsi/mpi3mr/mpi3mr.ko
-kernel/drivers/scsi/mpt3sas/mpt3sas.ko
 kernel/drivers/scsi/qedf/qedf.ko
 kernel/drivers/scsi/qla2xxx/qla2xxx.ko
-kernel/drivers/scsi/smartpqi/smartpqi.ko
 kernel/drivers/scsi/virtio_scsi.ko
 kernel/drivers/scsi/xen-scsifront.ko
 kernel/drivers/target/target_core_mod.ko
+kernel/drivers/thunderbolt/thunderbolt.ko
 kernel/drivers/ufs/core/ufshcd-core.ko
 kernel/drivers/ufs/host/ufshcd-pci.ko
 kernel/drivers/uio/uio.ko
@@ -255,6 +213,7 @@ kernel/drivers/usb/chipidea/ci_hdrc_tegra.ko
 kernel/drivers/usb/chipidea/ci_hdrc_usb2.ko
 kernel/drivers/usb/chipidea/usbmisc_imx.ko
 kernel/drivers/usb/class/cdc-acm.ko
+kernel/drivers/usb/class/cdc-wdm.ko
 kernel/drivers/usb/host/ehci-hcd.ko
 kernel/drivers/usb/host/ehci-orion.ko
 kernel/drivers/usb/host/ehci-pci.ko
@@ -265,10 +224,8 @@ kernel/drivers/usb/host/xhci-plat-hcd.ko
 kernel/drivers/usb/host/xhci-rcar-hcd.ko
 kernel/drivers/usb/host/xhci-tegra.ko
 kernel/drivers/usb/misc/sisusbvga/sisusbvga.ko
-kernel/drivers/usb/serial/ch341.ko
-kernel/drivers/usb/serial/cp210x.ko
-kernel/drivers/usb/serial/ftdi_sio.ko
-kernel/drivers/usb/serial/pl2303.ko
+kernel/drivers/usb/serial/option.ko
+kernel/drivers/usb/serial/usb_wwan.ko
 kernel/drivers/usb/storage/uas.ko
 kernel/drivers/usb/storage/ums-alauda.ko
 kernel/drivers/usb/storage/ums-cypress.ko
@@ -302,16 +259,13 @@ kernel/drivers/vhost/vhost_scsi.ko
 kernel/drivers/vhost/vhost_vdpa.ko
 kernel/drivers/vhost/vringh.ko
 kernel/drivers/virt/vmgenid.ko
-kernel/drivers/virtio/virtio_balloon.ko
 kernel/drivers/virtio/virtio_dma_buf.ko
-kernel/drivers/virtio/virtio_input.ko
-kernel/drivers/virtio/virtio_mmio.ko
-kernel/drivers/virtio/virtio_pci.ko
-kernel/drivers/virtio/virtio_pci_legacy_dev.ko
-kernel/drivers/virtio/virtio_pci_modern_dev.ko
 kernel/drivers/virtio/virtio_vdpa.ko
 kernel/drivers/watchdog/sbsa_gwdt.ko
+kernel/fs/binfmt_misc.ko
+kernel/fs/btrfs/btrfs.ko
 kernel/fs/fuse/virtiofs.ko
+kernel/fs/nfsd/nfsd.ko
 kernel/lib/objagg.ko
 kernel/lib/parman.ko
 kernel/lib/raid6/raid6_pq.ko
@@ -323,6 +277,24 @@ kernel/net/tls/tls.ko
 kernel/net/vmw_vsock/hv_sock.ko
 kernel/net/vmw_vsock/vmw_vsock_virtio_transport.ko
 kernel/net/xfrm/xfrm_interface.ko
+kernel/sound/core/snd-pcm-dmaengine.ko
+kernel/sound/core/snd-pcm.ko
+kernel/sound/core/snd-timer.ko
+kernel/sound/core/snd.ko
+kernel/sound/soc/codecs/snd-soc-hdmi-codec.ko
+kernel/sound/soc/snd-soc-core.ko
+kernel/sound/soundcore.ko
+modules.alias
+modules.alias.bin
 modules.builtin
+modules.builtin.alias.bin
+modules.builtin.bin
 modules.builtin.modinfo
+modules.dep
+modules.dep.bin
+modules.devname
 modules.order
+modules.softdep
+modules.symbols
+modules.symbols.bin
+modules.weakdep
-- 
2.51.1

